pkg/ddc/alluxio/shutdown.go

```diff
@@ -274,6 +279,14 @@ func (e *AlluxioEngine) destroyWorkers(expectedWorkers int32) (currentWorkers in
 			labelNames = append(labelNames, labelExclusiveName)
 		}
 
+		mutex.Lock()
+		currentDataset, err := strconv.Atoi(toUpdate.Labels[labelDatasetNum])
+		if err != nil {
+			return currentWorkers, err
+		}
+		toUpdate.Labels[labelDatasetNum] = strconv.Itoa(currentDataset - 1)
+		mutex.Unlock()
+
 		if len(toUpdate.Labels) < len(node.Labels) {
```

Comment:
Is there a situation where

The original labelDataSetNum was 6

1: 6 minus 1 is 5
2: update pending
3: 5 minus 1 is 4
4: update success, labelDatasetNum is: 4
5: the pending update success labelDatasetNum is: 5

Reply:
Hi @allenhaozi, it seems to me step 5 is going to fail because of resource version conflict. So I'm not getting when this situation would happen, can you explain it more?

Reply:
Because I see a lock mechanism on the change to the label
But it's not the same atomic operation as UPDATE

So if two updates come at the same time, will the result not be expected because of the instability of the network